contrib/src/databases.rs

```diff
@@ -0,0 +1,83 @@
+pub extern crate r2d2;
+
+use std::marker::{Send, Sized};
+use std::io;
+use std::error::Error;
+use rocket::config;
+use rocket::config::ConfigError;
+
+use self::r2d2::ManageConnection;
+
+#[cfg(feature = "diesel_sqlite")]
+pub extern crate diesel;
+
+#[cfg(feature = "pg_pool")]
+pub extern crate postgres;
+#[cfg(feature = "pg_pool")]
+pub extern crate r2d2_postgres;
+
+pub struct DatabaseConfig<'a> {
+    pub url: &'a str,
+    pub pool_size: u32,
+}
+
+pub fn database_config<'a>(name: &str, from: &'a config::Config) -> Option<DatabaseConfig<'a>> {
```

Comment:
My commit amendments accidentally deleted your comments, @SergioBenitez. I'm adding it in this comment to keep a record of it.

> @SergioBenitez 
> This is super hard to read. Don't be afraid to use ? to make this easier to follow.
>
> Also, why not store the &str in the DatabaseConfig? Something like:
> ```rust
> struct DatabaseConfig<'a> {
>     pub url: &'a str,
>     pub pool_size: u32,
> }
> ```

> @ELD
> The issue I ran into when I started writing this method was that the trait necessary to use the ? operator wasn't implemented, so I decided to use the monadic interface to chain together operations on a success type. It is somewhat convoluted. I'll see what other refactorings I can do on this (including attempting to use ? where possible).
>
> As for the second one, I'm not sure why I chose a String at the time. It was simple to switch it to a string slice, so I made I modified it to your example.
> 
> Additionally, on the topic of the DatabaseConfig struct, I want to extend it before we merge this to have all the extra options in key-value pairs like you mentioned in the original suggestion for a DatabaseConfig struct. The basic things necessary for a database connection is the url and the pool_size, but in the case of Rust Postgres, it takes an extra argument on the TLS mode of the connection. This complicates the procedural macro since we do have to discriminate based on the type of the connection since the interface differs from Diesel. Additionally, we need to keep track of those additional configuration options in this struct.

Reply:
In the future, I won't amend commits but instead make new ones. Not sure why I amended... Doesn't even make sense to amend pushed commits since it rewrites history and it'll all just get squashed at the end anyways.